src/Tarjan.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Graphp\Algorithms;
+
+use Fhaculty\Graph\Exception\InvalidArgumentException;
+use Fhaculty\Graph\Graph;
+use Fhaculty\Graph\Set\Vertices;
+use Fhaculty\Graph\Vertex;
+
+/**
+ * Finds the Strongly Connected Components in a Directed Graph, a graph component being said to be strongly connected
+ * if every vertex is reachable from every other vertex.
+ *
+ * More information here:
+ * @see https://en.wikipedia.org/wiki/Tarjan%27s_strongly_connected_components_algorithm
+ *
+ * This code was inspired by:
+ * @see http://github.com/Trismegiste/Mondrian/blob/master/Graph/Tarjan.php and
+ * @see https://code.google.com/p/jbpt/source/browse/trunk/jbpt-core/src/main/java/org/jbpt/algo/graph/StronglyConnectedComponents.java
+ */
+class Tarjan
+{
+    /** @var \SplObjectStorage  */
+    private $indexMap;
+
+    /** @var \SplObjectStorage  */
+    private $lowLinkMap;
+
+    /** @var Vertex[] */
+    private $stack;
+
+    /** @var int */
+    private $index;
+
+    /** @var Vertices[] */
+    private $partition;
+
+    /**
+     * Get the strongly connected components of this digraph by the Tarjan algorithm.
+     *
+     * @param Graph $graph Graph to analyze
+     * @throws InvalidArgumentException For undirected graph argument.
+     * @return Vertices[] Array of Strongly Connected components.
+     */
+    public function getStronglyConnectedVerticesFromDirectedGraph(Graph $graph)
```

Comment:
Naming this method is indeed a *tough challenge* :-)

Afaict this appears to be the first method so far which returns not only an array of vertex instances, but an array of vertices :open_mouth: 

Admittedly, I'm not a big fan of the current nor the suggested naming convention. The current suggested name matches this rather elaborate style, how about using something more concise?

Does naming this "TarjansAlgorithm::getConnectedComponents(Graph)" make sense to you?